src/components/ContactDetails/ContactDetailsAvatar.vue

```diff
@@ -80,6 +82,9 @@
 				<ActionButton v-if="!isReadOnly" icon="icon-picture" @click="selectFilePicker">
 					{{ t('contacts', 'Choose from files') }}
 				</ActionButton>
+				<ActionButton v-if="!isReadOnly && hasSocialId" icon="icon-sync" @click="getSocialAvatar">
+					{{ t('contacts', 'Get from social media') }}
+				</ActionButton>
```

Comment:
There should be one entry per compatible social sync handlers.
- ...
- Choose from files
- Get from Facebook
- Get from Twitter

Reply:
Wow, this needed quite some changes! Please take a look, I marked some lines with a FIXME comment where I felt it could be done better.